### PR TITLE
link removal on item removal and more access to OH-Definitions

### LINKED ISSUE #150: OpenHAB orphaned links after calling remove_item
While playing around with the links-endpoint I noticed that, when deleting an item that has links, those links get not deleted automatically. I found [this post in the OpenHAB community forum](https://community.openhab.org/t/little-orphan-item/86785/6) describing the issue.

Would it be a good idea to delete all links automatically before deleting the item in remove_item to avoid this?
(this would require looping through all links though as there is no possibility to get only the links for a specific item)

### PR DESCRIPTION
**Pull request by snpz**
Hi again.

I added a few more functions to directly access the OpenHAB Api and also implemented link-removal on item removal to close #150
I also added the get_thing() method someone [asked for in the OpenHAB forum](https://community.openhab.org/t/habapp-easy-automation-with-openhab/59669/139).

### CODE REVIEW

File: HABApp/openhab/oh_interface.py [hunk 1]
```diff
@@ -214,6 +214,49 @@ def get_item(self, item_name: str) -> OpenhabItemDefinition:
...
+        
+        :return: a list of all ItemChannelLinkDefinitions in OpenHAB
+        """
+
+        fut = asyncio.run_coroutine_threadsafe(
+            self.__connection.async_get_links(),
+            loop
+        )
+        return fut.result()
+
```
Comment (spacemanspiff2007) [inline HABApp/openhab/oh_interface.py:259]: Is there a use case where I want to display all links?

**Comment by snpz** [comment]
Sorry @spacemanspiff2007 I did not have much time the last week. How should we continue? Should I do a rebase of this PR first and then work on resolving conversations?
I saw you already added the get_thing() for example, I'm currently not sure with what to start.

There was also the open question regarding the BaseModel and the property names of ItemChannelLinkDefinition. I can't remember what the decision was unfortunately and I don't find the conversation.

**Comment by spacemanspiff2007** [comment]
No problem, I haven't had much time, too. I just wanted to push the release because it contained a bugfix for the time-of-day scheduler, now we can continue to develop as we have time.

**Comment by spacemanspiff2007** [comment]
This is stale so I am closing this

**Pull request closed without merging**